process/block/interceptedBlocks/interceptedTxBlock.go

```diff
@@ -0,0 +1,128 @@
+package interceptedBlocks
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// InterceptedTxBlockBody represents the wrapper over TxBlockBodyWrapper struct.
+type InterceptedTxBlockBody struct {
+	txBlockBody       block.Body
+	marshalizer       marshal.Marshalizer
+	hasher            hashing.Hasher
+	shardCoordinator  sharding.Coordinator
+	hash              []byte
+	isForCurrentShard bool
+}
+
+// NewInterceptedTxBlockBody creates a new instance of InterceptedTxBlockBody struct
+func NewInterceptedTxBlockBody(arg *ArgInterceptedTxBlockBody) (*InterceptedTxBlockBody, error) {
+	err := checkTxBlockBodyArgument(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	txBlockBody, err := createTxBlockBody(arg.Marshalizer, arg.TxBlockBodyBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	inTxBody := &InterceptedTxBlockBody{
+		txBlockBody:      txBlockBody,
+		marshalizer:      arg.Marshalizer,
+		hasher:           arg.Hasher,
+		shardCoordinator: arg.ShardCoordinator,
+	}
+	inTxBody.processFields(arg.TxBlockBodyBuff)
+
+	return inTxBody, nil
+}
+
+func createTxBlockBody(marshalizer marshal.Marshalizer, txBlockBodyBuff []byte) (block.Body, error) {
+	txBlockBody := make(block.Body, 0)
+	err := marshalizer.Unmarshal(&txBlockBody, txBlockBodyBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	return txBlockBody, nil
+}
+
+func (inTxBlkBdy *InterceptedTxBlockBody) processFields(txBuff []byte) {
+	inTxBlkBdy.hash = inTxBlkBdy.hasher.Compute(string(txBuff))
+
+	inTxBlkBdy.processIsForCurrentShard(inTxBlkBdy.txBlockBody)
+}
+
+func (inTxBlkBdy *InterceptedTxBlockBody) processIsForCurrentShard(txBlockBody block.Body) {
+	inTxBlkBdy.isForCurrentShard = false
+	for _, miniblock := range inTxBlkBdy.txBlockBody {
+		inTxBlkBdy.isForCurrentShard = inTxBlkBdy.miniblockForCurrentShard(miniblock)
```

Comment:
The interceptor only accepts the body for the next step: processing. In this next step only the useful miniblocks will be processed aka stored. If I will reject the whole body, the receiving shards will rely heavily on the request mechanism rather than gossip way of transmitting data because the incoming body will most of the time get rejected and the receiver shard will request missing miniblocks.